addons/udes_stock/models/stock_move_line.py

```diff
@@ -654,5 +657,15 @@ def _validate_location_dest(self):
         for picking in self.mapped('picking_id'):
             if not picking.is_valid_location_dest_id(location=location):
                 raise ValidationError(
-                    _("The location '%s' is not a child of the picking destination "
-                      "location '%s'" % (location.name, picking.location_dest_id.name)))
+                    _("The location '%s' is not a child of the picking "
+                      "destination location '%s'" %
+                      (location.name, picking.location_dest_id.name)))
+            if picking.picking_type_id.u_drop_location_constraint == 'enforce':
+                mls = self.filtered(lambda ml: ml.picking_id == picking)
+                locations = picking.get_suggested_locations(mls)
+                # location should be one of the suggested locations if there
+                # are any
+                if locations and location not in locations:
+                    raise ValidationError(
+                        _("Drop off location must be one of the suggested "
+                          "locations."))
```

Comment:
Nitpick: not using full stop elsewhere for error messages. Can be addressed later.